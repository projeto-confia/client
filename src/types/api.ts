import { ImageProps } from './image'
import { Nullable } from './nullable'
import { SocialNetworkType } from './socialNetwork'

export type ButtonLinkProps = {
  label: string
  url: string
}

export type LandingPageHeaderProps = {
  title: string
  description: string
  button: ButtonLinkProps
  image: ImageProps
}

export type ProcessStepProps = {
  label: string
  description: string
  icon: ImageProps
}

export type SectionHowWorksProps = {
  title: string
  processSteps: ProcessStepProps[]
}

export type SectionAboutProps = {
  title: string
  subtitle: string
  description: string
  image: ImageProps
}

export type SponsorProps = {
  name: string
  logo: ImageProps
}

export type SectionSponsorsProps = {
  title: string
  sponsors: SponsorProps[]
}

export type SocialNetworkLinkProps = {
  socialNetwork: SocialNetworkType
  url: string
}

export type MemberProps = {
  name: string
  role: string
  photo: ImageProps
  socialNetworkLinks: Nullable<SocialNetworkLinkProps[]>
}

export type SectionTeamProps = {
  title: string
  members: MemberProps[]
}

export type StatisticDataProps = {
  title: string
  subtitle: string
  count: number
}

export type SectionStatisticsProps = {
  statisticDatas: StatisticDataProps[]
}

export type LandingPageProps = {
  logo: ImageProps
  header: LandingPageHeaderProps
  sectionStatistics: SectionStatisticsProps
  sectionHowWorks: SectionHowWorksProps
  sectionAbout: SectionAboutProps
  sectionSponsors: SectionSponsorsProps
  sectionTeam: SectionTeamProps
}
