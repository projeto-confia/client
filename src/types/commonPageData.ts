export type CommonPageDataProps = {
  tabTitle: string
}
