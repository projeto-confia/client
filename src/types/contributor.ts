import { ImageProps } from './image'
import { Nullable } from './nullable'
import { SocialNetworkLinkProps } from './socialNetworkLink'

export type ContributorProps = {
  id: string
  name: string
  role: string
  description: string
  photo: ImageProps
  socialNetworkLinks: Nullable<SocialNetworkLinkProps[]>
}
