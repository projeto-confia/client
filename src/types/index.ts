export * from './homePage'
export * from './nullable'
export * from './image'
export * from './socialNetwork'
export * from './publicationsPage'
export * from './publication'
export * from './sectionStatistics'
export * from './sectionOverview'
export * from './projectPage'
export * from './videoCard'
export * from './eventsPage'
