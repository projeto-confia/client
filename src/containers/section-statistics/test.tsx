import { render, screen } from '@testing-library/react'
import { SectionStatisticsProps } from '../../types/index'

import SectionStatistics from '.'

const DEFAULT_PROPS: SectionStatisticsProps = {
  statisticDatas: [
    {
      title: 'Notícias revisadas',
      subtitle: 'Total de notícias suspeitas revisadas pelo confia',
      count: 57895,
    },
    {
      title: 'Enviadas para agências de checagem',
      subtitle: '',
      count: 22300,
    },
    {
      title: 'Fake news encontradas',
      subtitle:
        'Total de fake news encontradas e confirmadas pelas agências de checagem',
      count: 2400,
    },
  ],
}

describe('<SectionStatistics />', () => {
  it('should render all statisticData', () => {
    const { container } = render(<SectionStatistics {...DEFAULT_PROPS} />)

    expect(screen.getAllByRole('heading')).toHaveLength(3)

    expect(container.firstChild).toMatchSnapshot()
  })
})
