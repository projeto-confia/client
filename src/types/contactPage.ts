import { CommonPageDataProps } from './commonPageData'
import { SocialNetworkLinkProps } from './socialNetworkLink'

export type ContactPageProps = {
  commonPageData: CommonPageDataProps
  title: string
  email: string
  socialNetworkLinks: SocialNetworkLinkProps[]
}
