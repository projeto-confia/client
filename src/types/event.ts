import { CommonPageDataProps } from './commonPageData'
import { MediaProps } from './media'

export type EventProps = {
  commonPageData: CommonPageDataProps
  slug: string
  title: string
  subtitle?: string
  media: MediaProps
  content: string
}
