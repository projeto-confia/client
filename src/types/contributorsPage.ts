import { CommonPageDataProps } from './commonPageData'
import { ContributorProps } from './contributor'

export type ContributorsPageProps = {
  commonPageData: CommonPageDataProps
  title: string
  members: ContributorProps[]
}
