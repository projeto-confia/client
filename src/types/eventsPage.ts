import { CommonPageDataProps } from './commonPageData'
import { EventProps } from './event'

export type EventsPageProps = {
  commonPageData: CommonPageDataProps
  title: string
  events: EventProps[]
}
