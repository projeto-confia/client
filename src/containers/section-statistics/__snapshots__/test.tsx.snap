// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionStatistics /> should render all statisticData 1`] = `
<section
  class="MuiGrid-root MuiBox-root MuiBox-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiBox-root MuiBox-root-2 MuiPaper-elevation8 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root MuiBox-root MuiBox-root-3"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Notícias revisadas
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Total de notícias suspeitas revisadas pelo confia
        </p>
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          <p
            class="MuiTypography-root MuiTypography-h2 MuiTypography-colorPrimary"
          >
            57,895
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiBox-root MuiBox-root-5 MuiPaper-elevation8 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root MuiBox-root MuiBox-root-6"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Enviadas para agências de checagem
        </h2>
        
        <div
          class="MuiBox-root MuiBox-root-7"
        >
          <p
            class="MuiTypography-root MuiTypography-h2 MuiTypography-colorPrimary"
          >
            22,300
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiBox-root MuiBox-root-8 MuiPaper-elevation8 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root MuiBox-root MuiBox-root-9"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Fake news encontradas
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Total de fake news encontradas e confirmadas pelas agências de checagem
        </p>
        <div
          class="MuiBox-root MuiBox-root-10"
        >
          <p
            class="MuiTypography-root MuiTypography-h2 MuiTypography-colorPrimary"
          >
            2,400
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
